lib/Sema/PlaygroundTransform.cpp

```diff
@@ -322,6 +322,10 @@ class Instrumenter : InstrumenterBase {
       for (Decl *Member : NTD->getMembers()) {
         transformDecl(Member);
       }
+    } else if (auto *VD = dyn_cast<VarDecl>(D)) {
```

Comment:
Will this cast cover `lazy var` as well? For example, will we see results for `didSet` in:
```
struct S {
    lazy var v = 0 {
        didSet {
            // Do something
        }
    }
}

var s = S()
s.v = 1
```

Reply:
Great question, but I did check and we do see it (as long as there is something that generates a result in the `didSet()`).  We get `scope_entry` and `scope_exit` events for the code as written, and adding for example `let foo = 42` results in a `log_with_id` as well.

Reply:
Great, thanks for checking! Not sure what else could fall into the 'var-like, but not really a `VarDecl`' category

Reply:
I'm not sure either, but I will look into that.  I guess my feeling here is that this fix addresses known cases, and that we can address further fixes (with additional specific unit tests) with follow-on fixes.  My concern with making the check be too broad without working through the possibilities is that we might transform cases we don't want to.

I'll take a closer look at whether there is another type than `VarDecl` that we should be checking against, and with any luck perhaps one of the Swift engineers can provide some pointers.

Reply:
@abertelrud I think you want `AbstractStorageDecl` here instead of `VarDecl` because that would also cover subscripts and you should add tests for that as well.

Reply:
Great, thank you @xedin!  I'll update the PR.